code/modules/telesci/telepad.dm

```diff
@@ -51,18 +51,33 @@
 
 	return ..()
 
+var/global/list/cargopads = list() // Global List of Cargo Pads
 
 //CARGO TELEPAD//
 /obj/machinery/telepad_cargo
 	name = "cargo telepad"
-	desc = "A telepad used by the Rapid Crate Sender."
+	desc = "A telepad used by the cargo transporter."
 	icon = 'icons/obj/telescience.dmi'
-	icon_state = "pad-idle"
+	icon_state = "pad-idle-o"
 	anchored = 1
 	use_power = 1
 	idle_power_usage = 20
 	active_power_usage = 500
 	var/stage = 0
+	var/active = 0
+
+/obj/machinery/telepad_cargo/New()
+	..()
+	if (name == "cargo telepad")
+		name += " ([rand(100,999)])"
+	if (active && !cargopads.Find(src))

```

Comment:
Please use the list helpers when accessing lists, `Find` is rather slow and there are better solutions as seen here: https://github.com/tgstation/tgstation/blob/c30d07f55a1078b6463b56046884fbce9bac6959/code/__HELPERS/lists.dm